cpp/kiss_icp/core/Registration.cpp

```diff
@@ -68,7 +70,8 @@ constexpr double ESTIMATION_THRESHOLD_ = 0.0001;
 std::tuple<Eigen::Matrix6d, Eigen::Vector6d> BuildLinearSystem(
     const std::vector<Eigen::Vector3d> &source,
     const std::vector<Eigen::Vector3d> &target,
-    double kernel) {
+    double kernel,
+    int max_threads = tbb::task_arena::automatic) {
```

Comment:
I found this const on the `tbb` library which is set to `-1`. This could be the default everywhere, and if the user requires more control over the threads allow to change it to `1`, or `N`, or even `-1`